### PR TITLE
Add suite2p interface

### PR DESCRIPTION
**Pull request by alessandratrapani**
Add Suite2p interface

- [x] map segmentation plane and imaging plane names
- [x] add NWB converter

### CODE REVIEW

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py [hunk 1]
```diff
@@ -1,71 +1,78 @@
...
+    data_dir_path: Union[str, Path], output_dir_path: Union[str, Path], session_id: str, stub_test: bool = False
+):
     data_dir_path = Path(data_dir_path)
     output_dir_path = Path(output_dir_path)
     if stub_test:
         output_dir_path = output_dir_path / "nwb_stub"
     output_dir_path.mkdir(parents=True, exist_ok=True)
-
-    session_id = "subject_identifier_usually"
+    subject_id = "unknown"  # TODO ask for subject_id
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py]: you could add this as a function argument to `session_to_nwb()` if the source_data doesn't include this info

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py [hunk 2]
```diff
@@ -1,71 +1,78 @@
...
-    )
-    date = datetime.datetime.today()  # TO-DO: Get this from author
-    metadata["NWBFile"]["session_start_time"] = date
 
     # Update default metadata with the editable in the corresponding yaml file
     editable_metadata_path = Path(__file__).parent / "abdeladim_2023_metadata.yaml"
     editable_metadata = load_dict_from_file(editable_metadata_path)
     metadata = dict_deep_update(metadata, editable_metadata)
-
+    metadata = dict_deep_update(metadata, {"Subject": {"subject_id": subject_id}})
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py]: I think there is no need to call `dict_deep_update` to add the subject metadata, also where do you update the session_id?
```suggestion
    metadata["Subject"].update(subject_id=subject_id)
```

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+class Abdeladim2023SinglePlaneImagingInterface(BaseImagingExtractorInterface):
+    """
+    Data Interface for writing imaging data for the MouseV1 to NWB file using Abdeladim2023SinglePlaneImagingExtractor.
+    """
+
+    Extractor = Abdeladim2023SinglePlaneImagingExtractor
+
+    @classmethod
+    def get_available_channels(cls, folder_path):
+        file_path = sorted(glob.glob(f"{folder_path}/*.tif"))
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py:25]: `natsort.natsorted` is better to make sure the file names are in the correct order

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        self,
+        folder_path: FolderPathType,
+        channel_name: str,
+        plane_name: str,
+        verbose: bool = True,
+    ):
+        self.channel_name = channel_name
+        self.plane_name = plane_name
+
+        file_path = sorted(glob.glob(f"{folder_path}/*.tif"))
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I would suggest this everywhere where you do this naming: `file_path` would indicate a single path or string, I would suggest to use `file_paths`

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                "emission_lambda": 581.0,#TODO check
+                "description": "Red channel of the microscope, 550/50 nm filter.",#TODO check
+            },
+        }[channel_name_without_space]
+
+        optical_channel_metadata = channel_metadata
+
+        device_name = "ADD_DEVICE_NAME" #TODO add device name
+        metadata["Ophys"]["Device"][0].update(
+            name=device_name, description="ADD_DEVICE_DESCRIPTION", manufacturer="ADD_DEVICE_MANUFACTURER" #TODO add device description and manufacturer
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I think manufacturer is an optional argument in Device, if this microscope is custom built and you don't find this info in the papers then I think it's fine to not specify. just add a description that it is custom built

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py (hunk 2 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023_convert_session.py]: Ops, I forgot to update session_id. Thanks

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        )
+
+        two_photon_series_metadata = metadata["Ophys"]["TwoPhotonSeries"][0]
+        two_photon_series_metadata.update(
+            name=f"TwoPhotonSeries{channel_name_without_space}Plane{self.plane_name}",
+            imaging_plane=imaging_plane_name,
+            scan_line_rate=1/float(self.image_metadata['SI.hRoiManager.linePeriod']),
+            rate=self.imaging_extractor.get_sampling_frequency(),
+            description="ADD_2PSERIES_DESCRIPTION", #TODO ADD TwoPhotonSeries description
+            unit="px",
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]:
```suggestion
            unit="n.a.",
```

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            dimension = self.imaging_extractor.get_image_size()
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: just make sure dimension order should be width by height (x, y)

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imagingextractor.py [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+    is_writable = True
+    mode = "folder"
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        channel_name: Optional[str] = None,
+    ) -> None:
+        self.folder_path = Path(folder_path)
+        tif_file_paths = sorted(glob.glob(f"{self.folder_path}/*.tif"))
```
Comment (weiglszonja) [inline src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imagingextractor.py]: You can do Path(folder_path).glob(...) no need to import glob

**Comment by weiglszonja** [comment]
Overall LGTM, I added a few suggestions and comments. If you have the converter up and running, would be a good time to share a stubbed version of a dataset that contains the imaging and segmentation to check that the metadata was linked correctly, checking the image dimension etc. 

Also it is a bit hard at first what are the changes that you have here since #4 and probably #5 has the imaging part and this would be more about the converter and setting the metadata links.

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 3 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py:25]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 4 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 5 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 6 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 7 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: Yes, it should be correct

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imagingextractor.py (hunk 8 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imagingextractor.py]: I will change this in #4

**Comment by alessandratrapani** [comment]
Thank you so much @weiglszonja ! Implementing the corrections now

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 3 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py:25]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 4 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 5 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 6 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: I will change this in #5

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py (hunk 7 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py]: Yes, it should be correct

File: src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imagingextractor.py (hunk 8 above)
Comment (alessandratrapani) [reply src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imagingextractor.py]: I will change this in #4

**Pull request merged**